pkg/network/setup/controller.go

```diff
@@ -80,3 +93,230 @@ func (c *Controller) SetupCompleted(vmi *v1.VirtualMachineInstance) bool {
+			if interfaceStatus, exists := domainInterfaceStatusByMac[interfaceMAC]; exists {
+				newInterface.InterfaceName = interfaceStatus.InterfaceName
+				// Do not update if interface has Masquerede binding
+				// virt-controller should update VMI status interface with Pod IP instead
+				if !isForwardingBindingInterface {
+					newInterface.IP = interfaceStatus.Ip
+					newInterface.IPs = interfaceStatus.IPs
+				}
+				delete(domainInterfaceStatusByMac, interfaceMAC)
+			}
+			newInterfaces = append(newInterfaces, newInterface)
+		}
+
+		// If any of domain.Status.Interfaces were not handled above, it means that the vm contains additional
+		// interfaces not defined in domain.Spec.Devices.Interfaces (most likely added by user on VM or a SRIOV interface)
+		// Add them to vmi.Status.Interfaces
+		setMissingSRIOVInterfacesNames(existingInterfacesSpecByName, domainInterfaceStatusByMac)
+		for interfaceMAC, domainInterfaceStatus := range domainInterfaceStatusByMac {
+			newInterface := v1.VirtualMachineInstanceNetworkInterface{
+				Name:          domainInterfaceStatus.Name,
+				MAC:           interfaceMAC,
+				IP:            domainInterfaceStatus.Ip,
+				IPs:           domainInterfaceStatus.IPs,
+				InterfaceName: domainInterfaceStatus.InterfaceName,
+			}
+			newInterfaces = append(newInterfaces, newInterface)
+		}
+		vmi.Status.Interfaces = newInterfaces
+	}
+	return nil
+}
+
+func (c *Controller) getPodInterfacefromFileCache(vmi *v1.VirtualMachineInstance, ifaceName string) (*cache.PodCacheInterface, error) {
+	// Once the Interface files are set on the handler, they don't change
+	// If already present in the map, don't read again
+	podInterface, exists := c.podInterfaceVolatileCache.Load(vmi.UID, ifaceName)
+
+	if exists {
+		return podInterface, nil
+	}
+
+	//FIXME error handling?
+	podInterface, _ = c.ifaceCacheFactory.CacheForVMI(vmi).Read(ifaceName)
+
+	c.podInterfaceVolatileCache.Store(vmi.UID, ifaceName, podInterface)
+
+	return podInterface, nil
+}
+
+func setMissingSRIOVInterfacesNames(interfacesSpecByName map[string]v1.Interface, interfacesStatusByMac map[string]api.InterfaceStatus) {
+	for name, ifaceSpec := range interfacesSpecByName {
+		if ifaceSpec.SRIOV == nil || ifaceSpec.MacAddress == "" {
+			continue
+		}
+		if domainIfaceStatus, exists := interfacesStatusByMac[ifaceSpec.MacAddress]; exists {
+			domainIfaceStatus.Name = name
+			interfacesStatusByMac[ifaceSpec.MacAddress] = domainIfaceStatus
+		}
+	}
+}
+
+type PodInterfaceByVMIAndName struct {
+	syncMap sync.Map
+}
+
+func (p *PodInterfaceByVMIAndName) DeleteAllForVMI(vmiUID types.UID) {
+	// Clean Pod interface cache from map and files
+	p.syncMap.Range(func(key, value interface{}) bool {
+		if strings.Contains(key.(string), string(vmiUID)) {
+			p.syncMap.Delete(key)
+		}
+		return true
+	})
+}
+
+func (p *PodInterfaceByVMIAndName) Load(vmiUID types.UID, interfaceName string) (*cache.PodCacheInterface, bool) {
```

Comment:
What's the purpose of the boolean being returned ? Isn't the pointer alone sufficient ?